.evergreen/scripts/compile-unix.sh

```diff
@@ -216,3 +216,6 @@ build_dir="cmake-build"
 
 # Also validate examples.
 "${cmake_binary}" --build "${build_dir:?}" --target mongo_c_driver_examples
+
+# Check public headers for extra warnings.
+"${cmake_binary}" --build "${build_dir:?}" --target public-header-warnings
```

Comment:
```suggestion
if [[ "$EXTRA_CONFIGURE_FLAGS" != *"ENABLE_MONGOC=OFF"* ]]; then
  # Check public headers for extra warnings.
  "${cmake_binary}" --build "${build_dir:?}" --target public-header-warnings
fi
```

Suggestion may fix the [install-libmongoc-after-libbson](https://spruce.mongodb.com/task/mongo_c_driver_smoke_install_libmongoc_after_libbson_patch_e2b54b1be31f795b6d238c740ff1388679917ad7_670543f299a8ed000756ab50_24_10_08_14_38_45/logs?execution=0) task. The failure logs:

```
gmake: *** No rule to make target 'public-header-warnings'.  Stop.
```

That task first installs libbson without libmongoc by configuring with `ENABLE_MONGOC=OFF`.

`src/CMakeLists.txt` is only loaded if `ENABLE_MONGOC` is true [here](https://github.com/mongodb/mongo-c-driver/blob/060a759c228b63f09908a6b9a508c4447f89b911/CMakeLists.txt#L562).

To reproduce locally:
```
% cmake -DENABLE_MONGOC=OFF -S. -Bcmake-build
% cmake --build cmake-build --target public-header-warnings
make: *** No rule to make target `public-header-warnings'.  Stop.
```